src/xo-app/settings/acls/index.js

```diff
@@ -140,34 +141,56 @@ export default class Acls extends Component {
   constructor (props) {
     super(props)
     this.state = {
-      isAllSelected: {},
-      subjects: [],
       objects: [],
-      role: undefined
+      role: undefined,
+      subjects: [],
+      typeFilters: {}
     }
   }
 
-  _handleSelectObjects = objects => this.setState({objects})
-  _handleSelectRole = action => this.setState({action})
-  _handleSelectSubject = subjects => this.setState({subjects})
-
-  _toggleAll = type => {
-    const { isAllSelected, objects } = this.state
-    let newObjects
-    if (!isAllSelected[type]) {
-      newObjects = [ ...objects, ...toArray(createGetObjectsOfType(type)(store.getState())) ]
-    } else {
-      newObjects = filter(objects, object => object.type !== type)
+  _toggleTypeFilter = type => {
+    const {
+      someTypeFilters,
+      typeFilters,
+      objects
+    } = this.state
+
+    const newTypeFilters = { ...typeFilters, [type]: !typeFilters[type] }
+    const newSomeTypeFilters = some(newTypeFilters)
+
+    // If some objects need to be removed from the selected objects
+    if (!newTypeFilters[type] || !someTypeFilters && newSomeTypeFilters) {
+      this.setState({
+        objects: filter(objects, ({ type }) => !newSomeTypeFilters || newTypeFilters[type])
+      })
```

Comment:
Update `GenericSelect` to handle this case.

- use `@autoControlledInput`
- do not display selected options which are not available